server/src/main/java/org/elasticsearch/search/fetch/subphase/StoredFieldsPhase.java

```diff
@@ -77,7 +68,7 @@ public FetchSubPhaseProcessor getProcessor(FetchContext fetchContext) {
                     Collection<String> fieldNames = sec.getMatchingFieldNames(field);
                     for (String fieldName : fieldNames) {
                         MappedFieldType ft = sec.getFieldType(fieldName);
-                        if (ft.isStored() == false) {
+                        if (ft.isStored() == false || sec.isMetadataField(fieldName)) {
```

Comment:
Given that StoredField is a private record, and we create it within this class, I think we should simply skip creating an instance for metadata fields, instead of creating it and later ignoring it?

Reply:
If the field is a metadata field we hit the if and execute `continue` which means we do not create the stored field and we do not add it to `storedFields`. So I think we are not creating an instance od `StoredField` for metadata fields. So it looks to me that we already skip the creation of the stored field. Am I wrong?

I guess one could argue that later the check on `storedField.isMetadataField == false` is not necessary because of this since all fields we are are not metadata fields, right?

Reply:
You are correct, I think we should now be able to remove the isMetadataField member from the private StoredField record. There is also a conditional to exclude the `_source` from retrieval, that can and should be now removed.